modules/tinymce/src/models/dom/test/ts/browser/table/command/TableDeleteRowTest.ts

```diff
@@ -80,4 +80,43 @@ describe('browser.tinymce.models.dom.table.command.TableDeleteRowTest', () => {
     editor.execCommand('mceTableDeleteRow');
     TinyAssertions.assertContent(editor, initalContent);
   });
+
+  /** Create `rows` number of `tr` elements, with `cols` number of `td` elements inside each.  */
+  const tr = (rows: number, cols: number): string[] =>
+    Arr.range(rows, (r) => '<tr>' + Arr.range(cols, (c) => `<td>${r}-${c}</td>`).join('') + '</tr>');
+
+  Arr.each([
+    {
+      label: 'no colgroup',
+      pathToBody: [ 0, 0 ],
+      createTable: (tbody: string) => `<table><tbody>${tbody}</tbody></table>`,
+    },
+    {
+      label: 'with colgroup',
+      pathToBody: [ 0, 1 ],
+      createTable: (tbody: string) => `<table><colgroup><col><col><col></colgroup><tbody>${tbody}</tbody></table>`,
+    }
+  ], ({ label, pathToBody, createTable }) => {
+    context(label, () => {
+      it('TINY-6309: Should place cursor in adjacent cell when deleting the first and last row.', () => {
+        const editor = hook.editor();
+        const originalTBody = tr(4, 3);
+        editor.setContent(createTable(originalTBody.join('')));
+
+        // Delete last row:
+        TinySelections.setCursor(editor, [ ...pathToBody, 3, 1, 0 ], 0);
+        editor.execCommand('mceTableDeleteRow');
+        TinyAssertions.assertCursor(editor, [ ...pathToBody, 2, 1, 0 ], 3);
+        TinyAssertions.assertContent(editor, createTable(originalTBody.slice(0, -1).join('')));
+
+        // Delete first row:
```

Comment:
I would probably have deleting the first row and deleting the last row as separate `it` block test cases

Reply:
I've now split them into separate `it` blocks